content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,382 @@
+    <Caption>Use caption text that shows new information and provides helpful context</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832267-e00f88bb-6a81-42e2-a29d-63cc58a70550.png" />
+    <Caption>Don’t use caption text that is redundant</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Validation message
+
+A single validation message may be displayed to provide helpful information for a user to complete their task. For example: explain why a value is invalid so they can correct it and submit the form.
+
+An individual checkbox or radio should not have its own validation message or style.
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065196-b0146ade-eccb-4a71-b1b6-f4a82b96c115.png" />
+    <Caption>Show a validation message for the group of inputs</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065199-6cfa4893-55cb-4f81-a7d6-4b6ff66339eb.png" />
+    <Caption>Don’t show a validation message for each input</Caption>
+  </Dont>
+</DoDontContainer>
+
+Information from the caption should not be repeated in the error message. Show the validation message and remove the caption. A validation message makes the field easier to spot when a user is scanning for invalid fields.
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832268-519ebea3-e989-455c-b182-58e991d78a25.png" />
+    <Caption>Hide a caption that is redundant with the validation message</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832269-4756f340-c24a-41bd-8fa9-db807607c874.png" />
+    <Caption>Don’t show a repetitive validation message</Caption>
+  </Dont>
+</DoDontContainer>
+
+For more information about form validation, see the [validation guidelines](#invalid-input).
+
+## Input methods
+
+### Open-ended text
+
+![Text input, open-ended autocomplete text input, plain text input, textarea](https://user-images.githubusercontent.com/2313998/168104196-e0400967-a455-4513-97b0-3005d7ee3d0c.png)
+
+Use an open-ended text field when the field does not have a list of possible values. If the input is able to suggest values, use [autocomplete](#autocomplete) to allow users to pick a value or enter their own.
+
+### A set of selectable options
+
+![Action menu, autocomplete input with limited options, checkbox group, radio group, select](https://user-images.githubusercontent.com/2313998/168104199-2990e43c-6204-4ef5-970c-c9babaf1196c.png)
+
+Show a set of selectable options when there is a finite number of possible values.
+
+## Structure
+
+Forms should have a structure that makes it easy for users to scan. Form controls that are stacked vertically and left-aligned are easier for sighted users to scan visually.
```

Comment:
In production, we have a few examples of forms breaking this rule (in `Theme preferences` the radio btns stack horizontally, for example).

The do/don't format reads to me as a _must_ do, whereas it seems like this might be a suggestion instead? 🤔

Reply:
@heyamie I think the goal with interface guidelines is to try and document the ideal pattern even if its not always true today. The theme preference example is a great callout! That code and design will need to be addressed based on this guideline. I wonder if we need some kind of explainer that these do/dont's might not be true in production at the time of documentation 🤔

Reply:
Great point about horizontal radio buttons. The intention of this "do/don't" is to discourage forms that flow into multiple columns for no reason. I think it's totally fine to have horizontal radio buttons or have two related text inputs like "First name" / "Last name" on the same line.

I'll update the "do/don't" images to be more clear.